dpe/src/commands/certify_key.rs

```diff
@@ -19,21 +19,50 @@ pub struct CertifyKeyCmd {
     pub label: [u8; DPE_PROFILE.get_hash_size()],
 }
 
+impl CertifyKeyCmd {
+    pub const ND_DERIVATION: u32 = 1 << 31;
+
+    // Uses non-deterministic derivation.
+    const fn uses_nd_derivation(&self) -> bool {
+        self.flags & Self::ND_DERIVATION != 0
+    }
+}
+
 impl<C: Crypto> CommandExecution<C> for CertifyKeyCmd {
     fn execute(&self, dpe: &mut DpeInstance<C>, locality: u32) -> Result<Response, DpeErrorCode> {
+        // Make sure the operation is supported.
+        if !dpe.support.nd_derivation && self.uses_nd_derivation() {
+            return Err(DpeErrorCode::InvalidArgument);
+        }
+
         let idx = dpe
             .get_active_context_pos(&self.handle, locality)
             .ok_or(DpeErrorCode::InvalidHandle)?;
+        let context = &dpe.contexts[idx];
 
         // Make sure the command is coming from the right locality.
-        if dpe.contexts[idx].locality != locality {
+        if context.locality != locality {
             return Err(DpeErrorCode::InvalidHandle);
         }
 
-        // Derive CDI and public key
-        let cdi = dpe.derive_cdi(idx)?;
-        let pub_key = C::derive_ecdsa_pub(DPE_PROFILE.alg_len(), &cdi, &self.label, b"ECC")
+        let algs = DPE_PROFILE.alg_len();
+        let priv_key = if self.uses_nd_derivation() {
+            dpe.contexts[idx].cached_priv_key.take().unwrap_or({
```

Comment:
I'm not sure of the expected behavior here, but we might want to use `unwrap_or_else()`. The difference being that if there's a cached private key, then we won't derive a new key. Right now we always generate a new key but simply discard it if there's a cached private key.